.github/workflows/full_kubeflow_integration_test.yaml

```diff
@@ -4,278 +4,174 @@ on:
-    - name: Install Certificate Manager
-      run: ./tests/gh-actions/install_cert_manager.sh
-    - name: Install Istio Service Mesh
-      run: ./tests/gh-actions/install_istio-cni.sh
-    - name: Install OAuth2 Proxy for Authentication
-      run: ./tests/gh-actions/install_oauth2-proxy.sh
-    - name: Install Kubeflow Istio Resources
-      run: kustomize build common/istio-cni-1-24/kubeflow-istio-resources/base | kubectl apply -f -
-    - name: Install KF Multi Tenancy
-      run: ./tests/gh-actions/install_multi_tenancy.sh
-    # Right now KFP also modifies user namespaces
-    - name: Deploy Kubeflow Pipeline Components
-      run: ./tests/gh-actions/install_pipelines.sh
-    - name: Install dex
+
+    - name: Setup Environment
+      run: |
+        ./tests/gh-actions/install_KinD_create_KinD_cluster_install_kustomize.sh
+        ./tests/gh-actions/install_kubectl.sh
+        kustomize build common/kubeflow-namespace/base | kubectl apply -f -
+
+    - name: Install Core Infrastructure
       run: |
-        echo "Installing Dex..."
+        ./tests/gh-actions/install_cert_manager.sh
+        ./tests/gh-actions/install_istio-cni.sh
+        ./tests/gh-actions/install_oauth2-proxy.sh
+        kustomize build common/istio-cni-1-24/kubeflow-istio-resources/base | kubectl apply -f -
+        ./tests/gh-actions/install_multi_tenancy.sh
         kustomize build ./common/dex/overlays/oauth2-proxy | kubectl apply -f -
 
-        echo "Waiting for pods in auth namespace to become ready..."
-        kubectl wait --for=condition=Ready pods --all --timeout=180s -n auth
-    - name: Install central-dashboard
+    - name: Install Central Dashboard and Pipelines
       run: |
         kustomize build apps/centraldashboard/upstream/overlays/kserve | kubectl apply -f -
+        ./tests/gh-actions/install_pipelines.sh
         kubectl wait --for=condition=Ready pods --all --all-namespaces --timeout 180s
-    - name: Create Kubeflow User Profile
-      run: kustomize build common/user-namespace/base | kubectl apply -f -
-    - name: Verify User Profile Setup
+
+    - name: Create User Profile
       run: |
-        # Wait for profile controller to process the request
+        kustomize build common/user-namespace/base | kubectl apply -f -
         sleep 60
-
-        # Verify profile resources are properly created
-        KF_PROFILE=kubeflow-user-example-com
-        kubectl -n $KF_PROFILE get pods,configmaps,secrets
-
-        # Verify minio secret exists (critical for ML pipelines)
         if ! kubectl get secret mlpipeline-minio-artifact -n $KF_PROFILE > /dev/null 2>&1; then
-            echo "Error: Secret mlpipeline-minio-artifact not found in namespace $KF_PROFILE"
             exit 1
         fi
-    - name: Set up Python Environment
-      uses: actions/setup-python@v4
-      with:
-        python-version: '3.12'
-    - name: Install Python Dependencies
+
+    - name: Install Jupyter and Notebook Components
```

Comment:
There should be a reusable script for that that is also used in the other GHA workflwo file for jupyter related stuff. CHeck there, there are many files that must be consistent in .github/workflows. do not focus just on the single large worflow file. There are 20 or so of them.

Reply:
KISS principle means Keep it simple and stupid. do not add abstraction where it is not needed.